### ISSUE TITLE
Work up a web interface

### ISSUE THREAD

**Issue by psathyrella** [psathyrella/partis#164]
I think our long-term support effort will be minimized if most folks are running this through a web interface or something on, what EC2s or whatever that we control. Hell, we can even charge for large data sets if we want.

**Comment by psathyrella** [comment]
This is related to, or more or less the answer to, #150.

**Comment by matsen** [comment]
This is great, and I totally agree. It would be awesome to also get a database of parameter inferences as a side-effect of this. 

Here are two suggestions, if I may:

* Galaxy has a nice package which automatically wraps a command line tool just by using a variant argparse: https://github.com/erasche/gxargparse . This way we would get all of the Galaxy goodness for free. @bcclaywell is a Gxy pro, and could totally help.
* I think it would be lovely to try to make an abstraction layer that would run containerized partis as a black box, with as many inputs and outputs common to other programs in as close as we can get to a standard. Then someone can wrap other similar tools and run everything easily via a drop-down menu. This would build on the work that Andrew did this summer with bioboxes.

You could even focus on the latter, and us on the former? We'd want Brian's input too on scale-up.

**Comment by psathyrella** [comment]
Yeah, happy to work on whichever part. As far as the database, definitely, we should maybe have a tick box that says it's ok if we throw their data into the averages.

**Comment by matsen** [comment]
Great. Think you could pick up where Andrew left off with https://github.com/matsengrp/bioboxpartis ? 

Actually, we should hold off on doing anything until we get the go-ahead from @bcclaywell , who can advise if we could run a biobox on Gxy, and how this strategy would work WRT scale-up.

**Comment by bcclaywell** [comment]
Galaxy supports running tools in Docker containers but I don't think there's any way to run them on the cluster as of yet. I'm sure I could get a container running on maersk or exmar to work with Galaxy but it wouldn't scale well.

**Comment by psathyrella** [comment]
I was kind of imagining things would be running on amazon somewhere... is that possible?

**Comment by bcclaywell** [comment]
Ah, okay! Yes, it's possible to spin up Galaxy instances on EC2 with [CloudMan](https://wiki.galaxyproject.org/CloudMan) and then add custom tools and whatnot. I don't know if their Amazon machine image comes with support for Docker but it looks like you get ssh/sudo access once the cluster is up.

**Comment by psathyrella** [comment]
Cool! Thanks for looking that up.
